### ISSUE TITLE
unchecked cast false positive

### ISSUE THREAD

**Issue by xenoterracide** [spotbugs/spotbugs#676]
`L D BC: Unchecked/unconfirmed cast from org.eclipse.jgit.lib.BaseRepositoryBuilder to org.eclipse.jgit.storage.file.FileRepositoryBuilder of return value in com.xenoterracide.gradle.plugin.SemVerPlugin.apply(Project)  At SemVerPlugin.java:[line 26]`

from what I can tell this chained call s properly generic such that it would fail at compile time...

```
package com.xenoterracide.gradle.plugin;

import org.eclipse.jgit.api.Git;
import org.eclipse.jgit.api.errors.GitAPIException;
import org.eclipse.jgit.errors.InvalidPatternException;
import org.eclipse.jgit.lib.Repository;
import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

/**
 * inspired by @see <a href="https://github.com/cinnober/semver-git/">Cinnober's SemVer Git</a>
 */
public class SemVerPlugin implements Plugin<Project> {
    private final Logger log = LoggerFactory.getLogger( SemVerPlugin.class );


    @Override
    public void apply( Project project ) {
        try {
            Repository repo = new FileRepositoryBuilder()
                    .readEnvironment()
                    .findGitDir( project.getProjectDir() )
                    .build();

            Git git = new Git( repo );
            String version = git.describe().setMatch( "*[0-9].[0-9]*.[0-9]*" ).call();
            project.setVersion( version );
        }
        catch ( IOException | InvalidPatternException | GitAPIException e ) {
            log.error( "", e );
        }
    }
}
```

**Comment by zengleo** [comment]
Hi, I'd like to work on this issue for academic purposes.  Thanks!

**Comment by gtoison** [comment]
I _think_ that one was fixed by #2811
Please let me know if the problem persists
